driver/main.c

```diff
@@ -1275,8 +1276,49 @@ static inline void record_drop_x(struct ppm_consumer_t *consumer, struct timespe
 	}
 }
 
-static inline int drop_event(struct ppm_consumer_t *consumer, enum ppm_event_type event_type, enum syscall_flags drop_flags, struct timespec *ts)
+static inline int drop_event(struct ppm_consumer_t *consumer,
+			     enum ppm_event_type event_type,
+			     enum syscall_flags drop_flags,
+			     struct timespec *ts,
+			     struct pt_regs *regs)
 {
+	unsigned long close_fd = -1;
+	struct files_struct *files;
+	struct fdtable *fdt;
+	bool close_return = false;
+
+	// It's annoying but valid for a program to make a large number of
+	// close() calls on nonexistent fds. That can cause driver cpu usage
+	// to spike dramatically, so drop close events if the fd is not valid.
+	if (consumer->dropping_mode) {
```

Comment:
This check needs to be before UF_NEVER_DROP. I updated the comment, let me know if it's clear now.